lib/ssh_scan/api.rb

```diff
@@ -162,8 +169,25 @@ class API < Sinatra::Base
         if worker_id.empty? || uuid.empty?
           return {"accepted" => "false"}.to_json
         end
+        results = JSON.parse(request.body.first).first
+        settings.current_avg = settings.current_avg +
+        (results['scan_duration_seconds'] - settings.current_avg) / settings.count_req
```

Comment:
You are assuming here that all the requests arrive at `:curr_time(:window_start_time)`, but what if the requests are uniformly distributed over the time period?